src/kubeconfigs/aks.ts

```diff
@@ -0,0 +1,113 @@
+import * as core from '@actions/core'
+import * as io from '@actions/io'
+import * as crypto from 'crypto'
+import * as path from 'path'
+import {runAzKubeconfigCommandBlocking} from './azCommands'
+import {kubeLogin} from '../utils'
+import {Method, parseMethod} from '../types/method'
+const RUNNER_TEMP: string = process.env['RUNNER_TEMP'] || ''
+export const KUBECONFIG_LOCATION: string = path.join(
+   RUNNER_TEMP,
+   `aks_kubeconfig_${Date.now()}`
+)
+
+/**
+ * Sets the context by pulling kubeconfig via az aks command
+ * @returns Promise for the resulting kubeconfig in string form
+ */
+export async function getAKSKubeconfig(): Promise<string> {
+   const AZ_USER_AGENT_ENV = 'AZURE_HTTP_USER_AGENT'
+   const AZ_USER_AGENT_ENV_PS = 'AZUREPS_HOST_ENVIRONMENT'
+
+   const resourceGroupName = core.getInput('resource-group', {required: true})
+   const clusterName = core.getInput('cluster-name', {required: true})
+   const adminInput: string = core.getInput('admin')
+   const admin: boolean = adminInput.toLowerCase() === 'true'
+   const subscription: string = core.getInput('subscription') || ''
+   const azPath = await io.which('az', true)
+
+   const method: Method | undefined = parseMethod(
+      core.getInput('method', {required: true})
+   )
+
+   if (method !== Method.SERVICE_PRINCIPAL) {
+      throw Error(
+         (method == Method.KUBECONFIG ? 'Kubeconfig ' : 'Service Account ') +
```

Comment:
we can support service account and kubeconfig for aks clusters.

https://github.com/Azure/k8s-set-context/blob/03c6fe6b8a42e3a2e46b653edf0be15f6c051928/src/kubeconfigs/default.ts#L17

This code in the original does it. We also support kubeconfig (they just copy and paste their kubeconfig in to their secrets). Neither is recommended though

The method enum should probably only handle the auth methods (service principal or service account)